keyboards/kbdfans/kbd6x/keymaps/konstantin/keymap.c

```diff
@@ -104,6 +84,36 @@ void led_set_keymap(uint8_t usb_led) {
     check_light_led(usb_led);
 }
 
+bool process_record_keymap(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+    case RGB_TOG ... RGB_SPD:
+        if (record->event.pressed) {
+            // Shift+Toggle = reset RGB
+            if (keycode == RGB_TOG && get_mods() & MOD_MASK_SHIFT) {
+                reset_light();
+                return false;
+            }
+            restore_light();
+        } else {
+            check_light();
+        }
+        break;
+
+    // Combined RCtrl and layer
+    case RCTRL:
```

Comment:
`LM(L_RCTRL, MOD_RCTL)` will do this, as well.

Reply:
I had originally done exactly that (see https://github.com/vomindoraan/qmk_firmware/pull/14503/commits/56c3efd59748f2d8994766701440a1d33d4632d9), but the problem with that approach is that `LM(L_RCTRL, MOD_RCTL)` actually sends LCtrl, not RCtrl. This is because the top bit of `MOD_RCTL`, which signifies that it's a right mod, gets cut off when packed into `LM`. So I reverted that change and opted to keep my implementation, because I specifically need RCtrl in certain situations (e.g. VirtualBox).

Reply:
Aah, fair point. Not an easy way to fix that.

Reply:
I'll add a comment explaining this, for posterity.